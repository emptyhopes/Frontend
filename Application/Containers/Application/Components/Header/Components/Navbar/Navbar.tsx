import styled from "styled-components";

import { useLocation } from "react-router";
import { Link } from "react-router-dom";

const StyledNavbar = styled.div`
  display: flex;
  align-items: center;
`;

const StyledUl = styled.ul`
  display: flex;
  padding: 0;

  & li + li {
    margin: 0 0 0 40px;
  }
`;

const StyledLi = styled.li`
  & a {
    padding: 10px 20px;
    border-radius: 20px;
    background-color: #ffffff;
  }

  &.active {
    & a {
      background-color: #c6d8ff;
    }
  }
`;

const StyledLink = styled(Link)`
  padding: 10px 20px;
  border-radius: 20px;
  background-color: #ffffff;
`;

const CheckRouteActive = (url: string, path: string, children: JSX.Element) => {
  if (url === path) {
    return <StyledLi className="active">{children}</StyledLi>;
  }

  return <StyledLi>{children}</StyledLi>;
};

const Navbar = () => {
  const location = useLocation();
  return (
    <>
      <StyledNavbar>
        <nav>
          <StyledUl>
            {CheckRouteActive(location.pathname, "/", <StyledLink to={"/"}>Home</StyledLink>)}
            {CheckRouteActive(location.pathname, "/users", <StyledLink to={"/users"}>Users</StyledLink>)}
            {CheckRouteActive(location.pathname, "/posts", <StyledLink to={"/posts"}>Posts</StyledLink>)}
            {CheckRouteActive(location.pathname, "/todos", <StyledLink to={"/todos"}>Todos</StyledLink>)}
          </StyledUl>
        </nav>
      </StyledNavbar>
    </>
  );
};

export { Navbar };
