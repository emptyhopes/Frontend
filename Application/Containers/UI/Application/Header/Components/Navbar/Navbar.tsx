import styled from "styled-components";

import { Link, useLocation } from "react-router-dom";

import { Theme } from "@/Application/Containers/UI/Application/Theme/Theme";

const StyledNavbar = styled.div`
  display: flex;
  align-items: center;
`;

const StyledUl = styled.ul`
  display: flex;
  padding: 0;

  & li + li {
    margin: 0 0 0 40px;
  }
`;

const StyledLi = styled.li`
  & a {
    padding: 10px 20px;
    border-radius: 20px;
    background-color: #ffffff;
    font-weight: 700;
  }

  & a:hover {
    color: #;
  }

  &.active {
    & a {
      background-color: #c0c0c0;
    }
  }
`;

const StyledLink = styled(Link)`
  padding: 10px 20px;
  border-radius: 20px;
  background-color: #ffffff;
`;

const StyledTheme = styled(Theme)`
  margin: 0 0 0 60px;
`;

const Navbar: React.FunctionComponent = () => {
  const location = useLocation();

  const GetElementLink = (url: string, children: React.ReactNode) => {
    if (location.pathname === url) {
      return <StyledLi className={"active"}>{children}</StyledLi>;
    }

    return <StyledLi>{children}</StyledLi>;
  };

  return (
    <>
      <StyledNavbar>
        <nav>
          <StyledUl>
            {GetElementLink("/", <StyledLink to={"/"}>Home</StyledLink>)}
            {GetElementLink("/users", <StyledLink to={"/users"}>Users</StyledLink>)}
            {GetElementLink("/posts", <StyledLink to={"/posts"}>Posts</StyledLink>)}
            {GetElementLink("/todos", <StyledLink to={"/todos"}>Todos</StyledLink>)}

            <StyledTheme />
          </StyledUl>
        </nav>
      </StyledNavbar>
    </>
  );
};

export { Navbar };
