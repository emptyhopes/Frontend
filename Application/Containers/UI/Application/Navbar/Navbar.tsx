import styled from "styled-components";

import { Link, useLocation } from "react-router-dom";

import { Theme } from "@/Application/Containers/UI/Details/Theme/Theme";
import { Language } from "@/Application/Containers/UI/Details/Language/Language";

const StyledNavbar = styled.div`
  display: flex;
  align-items: center;
`;

const StyledUl = styled.ul`
  display: flex;
  padding: 0;

  & li + li {
    margin: 0 0 0 40px;
  }
`;

const StyledLi = styled.li`
  & a {
    padding: 10px 20px;
    border-radius: 20px;
    background-color: #ffffff;
    font-weight: 700;
  }

  & a:hover {
    color: #;
  }

  &.active {
    & a {
      background-color: #c0c0c0;
    }
  }
`;

const StyledLink = styled(Link)`
  padding: 10px 20px;
  border-radius: 20px;
  background-color: #ffffff;
`;

const StyledTheme = styled(Theme)`
  margin: 0 0 0 30px;
`;

const Navbar: React.FunctionComponent = (): React.ReactElement => {
  const location = useLocation();

  const GetElementLink = (url: string, children: React.ReactNode) => {
    if (location.pathname === url) {
      return <StyledLi className={"active"}>{children}</StyledLi>;
    }

    return <StyledLi>{children}</StyledLi>;
  };

  return (
    <>
      <StyledNavbar>
        <nav>
          <StyledUl>
            {GetElementLink("/", <StyledLink to={"/"}>Home</StyledLink>)}
            {GetElementLink("/users", <StyledLink to={"/users"}>Users</StyledLink>)}
            {GetElementLink("/posts", <StyledLink to={"/posts"}>Posts</StyledLink>)}
            {GetElementLink("/todos", <StyledLink to={"/todos"}>Todos</StyledLink>)}
          </StyledUl>
        </nav>
        <StyledTheme />
        <Language />
      </StyledNavbar>
    </>
  );
};

export { Navbar };
