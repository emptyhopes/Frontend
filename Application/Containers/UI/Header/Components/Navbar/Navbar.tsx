import styled from "styled-components";

import { useLocation } from "react-router";
import { Link } from "react-router-dom";

const StyledNavbar = styled.div`
  display: flex;
  align-items: center;
`;

const StyledUl = styled.ul`
  display: flex;
  padding: 0;

  & li + li {
    margin: 0 0 0 40px;
  }
`;

const StyledLi = styled.li`
  & a {
    padding: 10px 20px;
    border-radius: 20px;
    background-color: #ffffff;
  }

  & a:hover {
    color: red;
  }

  &.active {
    & a {
      background-color: #c6d8ff;
    }
  }
`;

const StyledLink = styled(Link)`
  padding: 10px 20px;
  border-radius: 20px;
  background-color: #ffffff;
`;

const Navbar = () => {
  const location = useLocation();
  return (
    <>
      <StyledNavbar>
        <nav>
          <StyledUl>
            <StyledLi className={location.pathname === "/" ? "active" : ""}>
              <StyledLink to={"/"}>Home</StyledLink>
            </StyledLi>
            <StyledLi className={location.pathname === "/users" ? "active" : ""}>
              <StyledLink to={"/users"}>Users</StyledLink>
            </StyledLi>
            <StyledLi className={location.pathname === "/posts" ? "active" : ""}>
              <StyledLink to={"/posts"}>Posts</StyledLink>
            </StyledLi>
            <StyledLi className={location.pathname === "/todos" ? "active" : ""}>
              <StyledLink to={"/todos"}>Todos</StyledLink>
            </StyledLi>
          </StyledUl>
        </nav>
      </StyledNavbar>
    </>
  );
};

export { Navbar };
