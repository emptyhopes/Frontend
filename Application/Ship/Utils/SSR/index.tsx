import * as React from "react";
import * as ReactDOM from "react-dom";

import { Application } from "@/Application/Containers/Application/Application";

const Structures = <Application />;

const element = document.getElementById("root");

ReactDOM.hydrate(Structures, element);
