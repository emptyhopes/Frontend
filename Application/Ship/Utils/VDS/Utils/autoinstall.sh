#! /bin/bash

# Update and upgrade system
sudo apt-get update --yes && sudo apt-get upgrade --yes

# Remove not use packages
sudo apt-get autoremove --yes

# Install packages
sudo apt-get install git nginx certbot python3-certbot-nginx --yes

# SWAP

sudo fallocate -l 2048M /swapfile
sudo chmod 600 /swapfile
sudo mkswap /swapfile
sudo swapon /swapfile

sudo cp /etc/fstab /etc/fstab.back

echo "/swapfile none swap sw 0 0" | sudo tee -a /etc/fstab

# NVM ( node@latest and npm@latest )

curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.2/install.sh | bash

source ~/.nvm/nvm.sh

nvm install node

npm install --global npm@latest

# Install letsencrypt SSL certificate

sudo certbot
