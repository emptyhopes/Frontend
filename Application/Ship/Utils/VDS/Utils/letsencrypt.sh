#! /bin/bash

hostname="emptyhopes.ru"

sudo certbot certonly --nginx -d $hostname