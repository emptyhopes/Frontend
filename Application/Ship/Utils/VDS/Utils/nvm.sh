#! /bin/bash

temporary=$(mktemp --directory)

curl --silent --location https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.2/install.sh >"$temporary/install.sh"

bash "$temporary/install.sh"

source ~/.nvm/nvm.sh

nvm install node
