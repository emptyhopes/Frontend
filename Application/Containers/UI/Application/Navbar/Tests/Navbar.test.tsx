import { RenderTests } from "@/Application/Ship/Tests/Utils/RenderTests";

import { Navbar } from "@/Application/Containers/UI/Application/Navbar/Navbar";

test("Navbar", () => {
  const { container } = RenderTests(<Navbar />);

  expect(container).toBeInTheDocument();
  expect(container).toMatchSnapshot();
});
