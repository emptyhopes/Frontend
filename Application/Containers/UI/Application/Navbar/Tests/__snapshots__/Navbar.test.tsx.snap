// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar 1`] = `
.c6 {
  position: relative;
  display: inline-block;
  width: 40px;
  height: 20px;
  cursor: pointer;
}

.c7 {
  width: 0;
  height: 0;
  opacity: 0;
}

.c7:focus + span {
  box-shadow: 0 0 1px #2196f3;
}

.c7:checked + span {
  background-color: #2196f3;
}

.c7:checked + span::before {
  -webkit-transform: translateX(18px);
  -ms-transform: translateX(18px);
  transform: translateX(18px);
}

.c8 {
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: 34px;
  background-color: #cccccc;
  cursor: pointer;
  -webkit-transition: 0.2s;
  transition: 0.2s;
}

.c8::before {
  position: absolute;
  bottom: 3px;
  left: 4px;
  width: 14px;
  height: 14px;
  border-radius: 50%;
  background-color: #ffffff;
  content: "";
  -webkit-transition: 0.2s;
  transition: 0.2s;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c4 svg {
  width: 20px;
  height: 20px;
  margin: 0 15px;
}

.c9 {
  padding: 10px 20px;
  border: 1px solid #e0e3e7;
  border-radius: 10px;
  background: #ffffff;
  color: #3e5060;
}

.c9:hover {
  background: #f0f0f0;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0;
}

.c1 li + li {
  margin: 0 0 0 40px;
}

.c2 a {
  padding: 10px 20px;
  border-radius: 20px;
  background-color: #ffffff;
  font-weight: 700;
}

.c2 a:hover {
  color: #;
}

.c2.active a {
  background-color: #c0c0c0;
}

.c3 {
  padding: 10px 20px;
  border-radius: 20px;
  background-color: #ffffff;
}

.c5 {
  margin: 0 0 0 30px;
}

<div>
  <div
    class="c0"
  >
    <nav>
      <ul
        class="c1"
      >
        <li
          class="c2 active"
        >
          <a
            class="c3"
            href="/"
          >
            Home
          </a>
        </li>
        <li
          class="c2"
        >
          <a
            class="c3"
            href="/users"
          >
            Users
          </a>
        </li>
        <li
          class="c2"
        >
          <a
            class="c3"
            href="/posts"
          >
            Posts
          </a>
        </li>
        <li
          class="c2"
        >
          <a
            class="c3"
            href="/todos"
          >
            Todos
          </a>
        </li>
      </ul>
    </nav>
    <div
      class="c4 c5"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 48v48m0 320v48m147.08-355.08l-33.94 33.94M142.86 369.14l-33.94 33.94M464 256h-48m-320 0H48m355.08 147.08l-33.94-33.94M142.86 142.86l-33.94-33.94"
          fill="none"
          stroke-linecap="round"
          stroke-miterlimit="10"
          stroke-width="32"
        />
        <circle
          cx="256"
          cy="256"
          fill="none"
          r="80"
          stroke-linecap="round"
          stroke-miterlimit="10"
          stroke-width="32"
        />
      </svg>
      <label
        class="c6"
      >
        <input
          aria-label="switch"
          class="c7"
          type="checkbox"
        />
        <span
          class="c8"
        />
      </label>
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 118a22 22 0 01-22-22V48a22 22 0 0144 0v48a22 22 0 01-22 22zm0 368a22 22 0 01-22-22v-48a22 22 0 0144 0v48a22 22 0 01-22 22zm113.14-321.14a22 22 0 01-15.56-37.55l33.94-33.94a22 22 0 0131.11 31.11l-33.94 33.94a21.93 21.93 0 01-15.55 6.44zM108.92 425.08a22 22 0 01-15.55-37.56l33.94-33.94a22 22 0 1131.11 31.11l-33.94 33.94a21.94 21.94 0 01-15.56 6.45zM464 278h-48a22 22 0 010-44h48a22 22 0 010 44zm-368 0H48a22 22 0 010-44h48a22 22 0 010 44zm307.08 147.08a21.94 21.94 0 01-15.56-6.45l-33.94-33.94a22 22 0 0131.11-31.11l33.94 33.94a22 22 0 01-15.55 37.56zM142.86 164.86a21.89 21.89 0 01-15.55-6.44l-33.94-33.94a22 22 0 0131.11-31.11l33.94 33.94a22 22 0 01-15.56 37.55zM256 358a102 102 0 11102-102 102.12 102.12 0 01-102 102z"
        />
      </svg>
    </div>
    <button
      class="c9"
    >
      Русский
    </button>
    <button
      class="c9"
    >
      Английский
    </button>
  </div>
</div>
`;
